src/GraphQL/GraphQLExtensions.cs

```diff
@@ -281,7 +281,23 @@ public static string[] IsValidLiteralValue(this IGraphType type, IValue valueAst
 
             var scalar = (ScalarGraphType)type;
 
-            var parseResult = scalar.ParseLiteral(valueAst);
+            object parseResult = null;
+            try
+            {
+                if (scalar is ICanParseScalar parseable)
+                {
+                    if (parseable.CanParseLiteral(valueAst)) // it still can throw
+                        return Array.Empty<string>();
+                }
+                else
+                {
+                    parseResult = scalar.ParseLiteral(valueAst);
+                }
+            }
+            catch (Exception)
+            {
+                // do nothing, just do not throw from IsValidLiteralValue
+            }
 
             if (parseResult == null)
```

Comment:
All this should simply be:

```suggestion
            if (!scalar.CanParseLiteral(valueAst))
```

Reply:
Done